### ISSUE TITLE
new_http_archive has a correctness problem on Windows

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#3093]
### Description of the problem / feature request / question:

After modifying the BUILD file specified in a new_http_archive, the BUILD file in external directory won't get updated.

This is a not an issue on Unix, because it would be a symlink to BUILD file in your source tree.
But on Windows, we copy the file instead.

### If possible, provide a minimal example to reproduce the problem:
[eigen.BUILD](https://github.com/tensorflow/tensorflow/blob/master/third_party/eigen.BUILD)
```
pcloudy@pcloudy0-w MSYS ~/workspace/my_tests/new_http_archive_test
$ tree .
.
├── eigen.BUILD
└── WORKSPACE

0 directories, 2 files

pcloudy@pcloudy0-w MSYS ~/workspace/my_tests/new_http_archive_test
$ cat WORKSPACE
new_http_archive(
  name = "eigen_archive",
  urls = [
    "http://mirror.bazel.build/bitbucket.org/eigen/eigen/get/f3a22f35b044.tar.gz",
    "https://bitbucket.org/eigen/eigen/get/f3a22f35b044.tar.gz",
  ],
  sha256 = "ca7beac153d4059c02c8fc59816c82d54ea47fe58365e8aded4082ded0b820c4",
  strip_prefix = "eigen-eigen-f3a22f35b044",
  build_file = "eigen.BUILD",
)

pcloudy@pcloudy0-w MSYS ~/workspace/my_tests/new_http_archive_test
$ bazel build @eigen_archive//...
.
INFO: Analysed target @eigen_archive//:eigen.
INFO: Found 1 target...
Target @eigen_archive//:eigen up-to-date (nothing to build)
INFO: Elapsed time: 45.615s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action

pcloudy@pcloudy0-w MSYS ~/workspace/my_tests/new_http_archive_test
$ echo "blabla" >> eigen.BUILD

pcloudy@pcloudy0-w MSYS ~/workspace/my_tests/new_http_archive_test
$ bazel build @eigen_archive//...  # THIS SHOULD FAILED
INFO: Analysed target @eigen_archive//:eigen.
INFO: Found 1 target...
Target @eigen_archive//:eigen up-to-date (nothing to build)
INFO: Elapsed time: 0.632s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action
```

The same problem doesn't happen to `new_local_repository`

I'll look into this.

**Comment by abergmeier-dsfishlabs** [comment]
Could we have this fixed in 0.5.x?

**Comment by meteorcloudy** [comment]
I have created https://bazel-review.googlesource.com/#/c/11931/, but it's currently under review and have some problems to discuss. I guess 0.5.x is possible but not 0.5.2.
